tapchannel/aux_invoice_manager.go

```diff
@@ -367,6 +377,33 @@ func isAssetInvoice(invoice *lnrpc.Invoice, rfqLookup RfqLookup) bool {
 	return false
 }
 
+// validateAssetHTLC runs a couple of checks on the provided asset HTLC.
+func (s *AuxInvoiceManager) validateAssetHTLC(htlc *rfqmsg.Htlc) error {
+	rfqID := htlc.RfqID.ValOpt().UnsafeFromSome()
+
+	// Retrieve the asset identifier from the RFQ quote.
+	identifier, err := s.identifierFromQuote(rfqID)
+	if err != nil {
+		return fmt.Errorf("could not extract assetID from "+
+			"quote: %v", err)
+	}
+
+	// Check for each of the asset balances of the HTLC that the identifier
+	// matches that of the RFQ quote.
+	for _, v := range htlc.Balances() {
+		if !identifier.HasId() {
+			return fmt.Errorf("asset specifier has empty assetID")
+		}
+
+		if v.AssetID.Val != *identifier.UnwrapIdToPtr() {
+			return fmt.Errorf("mismatch between htlc asset ID " +
+				"and rfq asset ID")
+		}
```

Comment:
This feels very much against the whole `Option` paradigm... We could add an `ID()` or `IDOption()` method to the specifier that returns the underlying `fn.Option` directly, then we could do this the "intended" way:

```
		err := fn.MapOptionZ(
			identifier.IdOption(), func(id asset.ID) error {
				if v.AssetID.Val != id {
					return fmt.Errorf("mismatch between " +
						"htlc asset ID and rfq asset " +
						"ID")
				}

				return nil
			},
		)
		if err != nil {
			return err
		}
```

But non-blocking.